clang/test/Driver/sycl-dump-device-code.cpp

```diff
@@ -0,0 +1,23 @@
+// SYCL offloading tests using -fsycl-dump-device-code
+
+// Verify that -fsycl-dump-device-code puts the device code (.spv files)
+// in the user provided directory.
+
+// clang -fsycl -target x86_64-unknown-linux-gnu
+// RUN: %clang -fsycl -fno-sycl-instrument-device-code -fno-sycl-device-lib=all -target x86_64-unknown-linux-gnu -fsycl-dump-device-code=/user/input/path/ %s -### 2>&1 \
+// RUN: | FileCheck %s --check-prefixes=CHK-FSYCL-DUMP-DEVICE-CODE
+
+// clang -fsycl -fsycl-targets=spir64-unknown-unknown
+// RUN: %clang -fsycl  -fsycl-targets=spir64-unknown-unknown -target x86_64-unknown-linux-gnu -fsycl-dump-device-code=/user/input/path/ %s -### 2>&1 \
+// RUN: | FileCheck %s --check-prefixes=CHK-FSYCL-DUMP-DEVICE-CODE
+
+// clang --driver-mode=g++
+// RUN: %clangxx -fsycl  -fsycl-targets=spir64-unknown-unknown -target x86_64-unknown-linux-gnu -fsycl-dump-device-code=/user/input/path/ %s -### 2>&1 \
+// RUN: | FileCheck %s --check-prefixes=CHK-FSYCL-DUMP-DEVICE-CODE
+
+// CHK-FSYCL-DUMP-DEVICE-CODE: llvm-foreach{{.*}} "--in-file-list={{.+\.txt}}" "--in-replace={{.+\.txt}}" "--out-ext=spv" "--out-file-list={{.+\.txt}}" "--out-replace={{.+\.txt}}" "--out-dir=/user/input/path/"
```

Comment:
I wonder what `--out-dir` comes out to be on Windows, does it use `\\user\\input\\path`?

Reply:
`--out-dir` on Windows is `\\user\\input\\path\\`
I need to update the test to check for both Windows and Linux style of path using regular expression